src/DateTimePicker.jsx

```diff
@@ -0,0 +1,240 @@
+	/**
+	* @function getDate
+	* @description gets a date from the datetime value in state
+	* @return Date
+	*/
+	getDate() {
+		if (!this.state.datetime) {
+			return;
+		}
+		const datetime = new Date(this.state.datetime);
+		return new Date(datetime.getFullYear(), datetime.getMonth(), datetime.getDate());
+	}
+
+	/**
+	* @function setDate
+	* @description sets the date portion of the state's datetime
+	* @param Date value date to set in state
+	*/
+	setDate(value) {
+		const newDate = new Date(value);
+		const datetime = new Date(this.state.datetime);
+
+		datetime.setFullYear(newDate.getFullYear(), newDate.getFullMonth(), newDate.getDate());
+		this.setState({ datetime });
+	}
+
+	/**
+	* @function parseTimeFromDateTime
+	* @description gets the time string part of our datetime state
+	* @return String
+	*/
+	parseTimeFromDateTime(datetime) {
+		return (datetime.toTimeString().split(' ')[0]).split(':', 2).join(':'); // TODO localize toLocaleTimeString ?
+	}
+
+	/**
+	* @function getTime
+	* @description gets the time portion of the state's datetime
+	* @return String time portion of datetime  (hour and min)
+	*/
+	getTime() {
+		if (!this.state.datetime) {
+			return;
+		}
+		const datetime = new Date(this.state.datetime);
+		// split on space and leave off timezone and seconds
+		return this.parseTimeFromDateTime(datetime);
+	}
+
+	/**
+	* @function setTime
+	* @description sets the time portion of the state's datetime
+	* @param value String valid time string
+	*/
+	setTime(value) {
+		const newTime = new Date(value);
+		const datetime = new Date(this.state.datetime);
+
+		datetime.setTime(newTime.getTime());
+		this.setState({ datetime });
+	}
+
+	setDateTime(value) {
+		this.setState({ datetime: new Date(value) });
+	}
+
+	/**
+	* @function onFocus
+	* @param Event e
+	* @description called if datetime compound component and used to set focused
+	* class on combo component
+	*/
+	onFocus(e) {
+		this.backgroundEl.classList.add('focused');
+	}
+
+	/**
+	* @function onBlur
+	* @param Event e
+	* @description called when either of the date or time inputs is blurred
+	* takes off focus class when neither of them are in focus
+	*/
+	onBlur(e) {
+		const timeInput = ReactDOM.findDOMNode(this.timeComponent).querySelector('input[type=time]');
+		const dateInput = ReactDOM.findDOMNode(this.dateComponent).querySelector('input');
+		if (document.activeElement !== timeInput && document.activeElement !== dateInput) {
+			this.backgroundEl.classList.remove('focused');
+		}
+	}
+
+	render() {
+		const {
+			callback,	// eslint-disable-line no-unused-vars
+			className,
+			dateOnly,
+			datepickerOptions,	// eslint-disable-line no-unused-vars
+			id,
+			label,
+			value,	// eslint-disable-line no-unused-vars
+			required,
+			...other
+		} = this.props;
+
+		const classNames = cx(
+			'dateTimePicker',
+			{calendarTimeComponent : !dateOnly },
+			{required},
+			className
+		),
+			labelClassNames = cx({required}),
+			timeInputName = `${name}-time`;
```

Comment:
new `const` declaration for each new const is another option - I've stopped doing comma-delimited variable declarations entirely, and I feel like it has had a (small) benefit in scan-ability and clean diffs (no more managing commas-vs-semicolons).